rfcs/text/0016_buildkite.md

```diff
@@ -0,0 +1,907 @@
+      - [GitHub Integration](#github-integration-1)
+  - [What we will build and manage](#what-we-will-build-and-manage)
+    - [Elastic Buildkite Agent Manager](#elastic-buildkite-agent-manager)
+      - [Overview](#overview-1)
+      - [Design](#design)
+        - [Protection against creating too many instances](#protection-against-creating-too-many-instances)
+      - [Configuration](#configuration)
+      - [Build / Deploy](#build--deploy)
+    - [Elastic Buildkite PR Bot](#elastic-buildkite-pr-bot)
+      - [Overview](#overview-2)
+      - [Configuration](#configuration-1)
+      - [Build / Deploy](#build--deploy-1)
+    - [Infrastructure](#infrastructure)
+    - [Monitoring / Alerting](#monitoring--alerting)
+    - [Agent Image management](#agent-image-management)
+    - [Buildkite org-level settings management](#buildkite-org-level-settings-management)
+    - [IT Security Processes](#it-security-processes)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+  - [Jenkins](#jenkins)
+    - [Required](#required-2)
+      - [Scalable](#scalable-2)
+      - [Stable](#stable-2)
+        - [Updates](#updates)
+      - [Surfaces information intuitively](#surfaces-information-intuitively-2)
+      - [Pipelines](#pipelines-2)
+      - [Advanced Pipeline logic](#advanced-pipeline-logic-2)
+      - [Cloud-friendly pricing model](#cloud-friendly-pricing-model-2)
+      - [Public access](#public-access-2)
+      - [Secrets handling](#secrets-handling-2)
+      - [Support or Documentation](#support-or-documentation-2)
+      - [Scheduled Builds](#scheduled-builds-2)
+      - [Container support](#container-support-2)
+    - [Desired](#desired-2)
+      - [Customization](#customization-2)
+      - [Core functionality is first-party](#core-functionality-is-first-party-2)
+      - [First-class support for test results](#first-class-support-for-test-results-2)
+      - [GitHub Integration](#github-integration-2)
+  - [Other solutions](#other-solutions)
+    - [CircleCI](#circleci)
+    - [GitHub Actions](#github-actions)
+- [Adoption strategy](#adoption-strategy)
+- [How we teach this](#how-we-teach-this)
+- [Unresolved questions](#unresolved-questions)
+
+# Summary
+
+Implement a CI system for Kibana teams that is highly scalable, is stable, surfaces information in an intuitive way, and supports pipelines that are easy to understand and change.
+
+This table provides an overview of the conclusions made throughout the rest of this document. A lot of this is subjective, but we've tried to take an honest look at each system and feature, based on a large amount of research on and/or experience with each system, our requirements, and our preferences as a team. Your team would likely come to different conclusions based on your preferences and requirements.
+
+<!-- eslint-disable max-len -->
+
+|                                      | Jenkins | Buildkite | GitHub Actions | CircleCI |
```

Comment:
Should we add TeamCity to this comparison since we have a good understand of it's capabilities and limitations?

Reply:
I could probably do that fairly quickly. Would it be useful to have there if we know it's not an option?